controllers/v1/reports.js

```diff
@@ -0,0 +1,285 @@
+/**
+ * name : reports.js
+ * author : Rakesh
+ * created-date : 01-Nov-2020
+ * Description : Reports related information.
+ */
+
+// Dependencies
+const csv = require('csvtojson');
+const reportsHelper = require(MODULES_BASE_PATH + "/reports/helper");
+
+/**
+   * Reports
+   * @class
+*/
+
+module.exports = class Reports {
+
+    static get name() {
+        return "reports";
+    }
+
+    /**
+    * @api {post} /improvement-project/api/v1/reports/entity/:_id
+    * Entity Report.
+    * @apiVersion 1.0.0
+    * @apiGroup Reports
+    * @apiSampleRequest /improvement-project/api/v1/reports/entity/5f731631e8d7cd3b88ac0659
+    * @apiParamExample {json} Request:
+     {
+        "reportType":"lastMonth",
+        "programId":"5da5a3af6ee4a93ce5a1987a"
+      }
+    * @apiParamExample {json} Response:
+    * {
+    "message": "Reports generated successfully.",
+    "status": 200,
+    "result": {
+        "data_available": true,
```

Comment:
@rakeshSgr - Why is this key required? And should it not be the camel case?

Reply:
Rajneesh, requested this response.